### PR TITLE
Update tailwind config

### PR DESCRIPTION
**Pull request by damsfx**
Add theme's `content/*.htm` files to tailwind config.

### CODE REVIEW

**Comment by LukeTowers** [comment]
Why would we need to consider the Winter.Pages content when generating the styles?

**Comment by damsfx** [comment]
> Why would we need to consider the Winter.Pages content when generating the styles?

Well, static page can define some content with Tailwind classes. Two use cases :  

**[Custom page fields](https://github.com/wintercms/wn-pages-plugin/blob/main/docs/documentation.md#custom-page-fields)** :   
They can define `dropdown` fields with options related to tailwind classes.
```twig
padding:
    '': 'Default'
    'py-0': '0'
    'py-4': '4'
    'py-8': '8'
    'py-10': '10'
    'py-12': '12'
    'py-16': '16'
    'py-20': '20'
```

**[Blocks of Winter Blocks plugin](https://github.com/wintercms/wn-blocks-plugin)** :   
https://github.com/wintercms/wn-blocks-plugin/blob/4ea3447ab924de737290aadc272888c8a4ed0810/blocks/image.block#L20-L24
And 
https://github.com/wintercms/wn-blocks-plugin/blob/4ea3447ab924de737290aadc272888c8a4ed0810/blocks/image.block#L27

Without forgetting that the content can be generated from the backend of the site in `CMS/Content`.   

That's why I thought it would be useful to include this directory in the list of those considered by Tailwind by default.

**Comment by LukeTowers** [comment]
@damsfx if the layout provides a tailwindcss class for the pages plugin to use then you need to include it in the layout in such a way that the CSS compiler can detect it. The same applies for blocks. Thus, just including blocks and layouts should trigger tailwind to generate the necessary classes to support any actual content being rendered by those templates.

**Comment by damsfx** [comment]
@LukeTowers in the case of the blocks plugin : 
https://github.com/wintercms/wn-blocks-plugin/blob/4ea3447ab924de737290aadc272888c8a4ed0810/blocks/image.block#L20-L24
The theme's content folder contains pages with tailwind classes : 
image

I must have missed something in your explanation!? ¯\\_(ツ)_/¯

**Comment by LukeTowers** [comment]
@damsfx no worries, it took me a bit to figure out at first too.

My point is that Tailwind will be able to pick up the need for `w-full` to be generated simply by the inclusion of the `*.block` files in the asset compilation step, rather than needing the content files themselves to be included at that point.

When you include `*.block`, Tailwind will see the lines where those options are defined and treat them as an inclusion of that class that it needs to process (see https://github.com/wintercms/wn-blocks-plugin/blob/main/blocks/button_group.block#L27-L28). That means that all you should have to do is include the blocks folders and the layout folders and you're good to go; unless for some reason you have a field that allows for arbitrary content from the user. 

In some cases, you may need to do a slight workaround in your block file; i.e. for the example of a custom Froala dropdown where those dropdown values are defined elsewhere in the project (whether that's the DB or perhaps in some JS somewhere), then you could do the following:

```yaml
name: richtext
form:
	fields:
		content:
			buttons: [myCustomFontButton]
==
{#
    Include the following classes used by the content -> myCustomFontButton setting
    py-0
	py-2
	py-4
	... // etc
#}
<div>{{ data.content | raw }}</div>
```

Does that kind of make sense? Is there a better way that could be introduced or documented so it's easier to understand / implement?

**Comment by damsfx** [comment]
> That means that all you should have to do is include the blocks folders and the layout folders and you're good to go; unless for some reason you have a field that allows for arbitrary content from the user.
 
Indeed I have this type fields.  
I also have repeated field declarations for the static page plugin that use dropdown fields to choose a value.

> Does that kind of make sense? Is there a better way that could be introduced or documented so it's easier to understand / implement?

Yes it makes sense !

In my case, here are the paths that I systematically use in the Taliwind configuration: 
- `./configs/**/*.yaml`
- `./content/**/*.htm`
- `./layouts/**/*.htm`
- `./pages/**/*.htm`
- `./partials/**/*.htm`

And now I add the blocks defined in the plugins or the theme.

So I think we can close this RP.

**Comment by LukeTowers** [comment]
Sounds good, although I really would recommend not including your content folder to force you to ensure that it's setup to handle the styles being included without having to have your content itself being scanned for styles. That way you can have a more portable deployment that doesn't rely on all of the content being present in the repo when the build script is run to continue to work well.

File: tailwind.config.js [hunk 1]
```diff
@@ -2,9 +2,10 @@ const colors = require('tailwindcss/colors');
 
 module.exports = {
     content: [
+        './content/**/*.htm',
```
Comment (LukeTowers) [inline tailwind.config.js]:
```suggestion
        './blocks/**/*.block',
```

File: tailwind.config.js [hunk 2]
```diff
@@ -2,9 +2,10 @@ const colors = require('tailwindcss/colors');
 
 module.exports = {
     content: [
+        './blocks/**/*.block',
```
Comment (LukeTowers) [inline tailwind.config.js:6]:
```suggestion
        '../../plugins/*/*/blocks/*.block',
        './blocks/**/*.block',
```

**Pull request merged**